internal/services/resource/template_deployment_common.go

```diff
@@ -139,6 +140,25 @@ func filterOutTemplateDeploymentParameters(input interface{}) interface{} {
 	return output
 }
 
+func deleteResourceById(ctx context.Context, resourcesClient *resources.Client, resourceID string, resourceProviderApiVersion string) (result resources.DeleteByIDFuture, err error) {
```

Comment:
why are we creating a function when this is only used one place? could we please inline it?